### ISSUE TITLE
Eclipse Che start error

### ISSUE THREAD

**Issue by Cookie-Wang** [eclipse-che/che#180]
my environment:
java -version
java version "1.8.0_66"
Java(TM) SE Runtime Environment (build 1.8.0_66-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.66-b17, mixed mode)

JAVA_HOME is /usr/java/jdk1.8.0_66

the start error：

/opt/the/assembly-main/target/eclipse-che-4.0.0-beta-8-SNAPSHOT/eclipse-che-4.0.0-beta-8-SNAPSHOT/bin/che.sh start

!!! This Linux user is not in docker group. See https://docs.docker.com/engine/installation/ubuntulinux/#create-a-docker-group

Looks like something went wrong. Possible issues:
1. (Win | Mac) VirtualBox not installed          ==> Rerun Docker Toolbox installation
2. (Win | Mac) Docker Machine not installed      ==> Rerun Docker Toolbox installation
3. (Win | Mac) Docker is not reachable           ==> Docker VM failed to start
4. (Win | Mac) Docker ok, but docker ps fails    ==> Docker environment variables not set properly
5. (Linux) Docker is not reachable               ==> Install: wget -qO- https://get.docker.com/ | sh
6. (Linux) Permissions not properly set          ==> Che must run as UID 1000 with user in docker group
7. Could not find the Che app server             ==> Did /tomcat get moved away from CHE_HOME?
8. Wrong version of Java found                   ==> Che requires Java 1.8
9. Did you use the right parameter syntax?       ==> See usage

We have seen issues with VirtualBox on Windows where your VM gets corrupted when your computer is
suspended while the VM is still running. This will appear as SSH or ethernet connection issues. This is
rare, but if encountered, current known solution is to uninstall VirtualBox and Docker Toolbox, and then
reinstall.

Che Environment Variables:
  (REQUIRED) JAVA_HOME                             ==> Location of Java runtime
  (REQUIRED: WIN|MAC) DOCKER_TOOLBOX_INSTALL_PATH  ==> Location of Docker Toolbox
  (REQUIRED: WIN|MAC) VBOX_MSI_INSTALL_PATH        ==> Location of VirtualBox
  (OPTIONAL) CHE_HOME                              ==> Directory where Che is installed
  (OPTIONAL) CHE_LOCAL_CONF_DIR                    ==> Directory with custom Che .properties files
  (OPTIONAL) CHE_LOGS_DIR                          ==> Directory for Che output logs
  (OPTIONAL) CHE_DOCKER_MACHINE_NAME               ==> (Win | Mac) Name of VM created by docker-machine
  (OPTIONAL) DOCKER_MACHINE_HOST                   ==> (Linux) Docker host IP - set if browser clients remote

Usage:
  che [OPTIONS] [run | start | stop]
     -i,      --image        Launches Che within a Docker container using latest image
     -i:tag,  --image:tag    Launches Che within a Docker container using specific image tag
     -p:port, --port:port    Port that Che server will use for HTTP requests; default=8080
     -r:ip,   --remote:ip    If Che clients are not localhost, set to IP address of Che server
     -m:vm,   --machine:vm   For Win & Mac, sets the docker-machine VM name to vm; default=default
     -s,      --suppress     Do not print browser client connection information
     -h,      --help         Show this help
     -d,      --debug        Use debug mode (prints command line options + app server debug)
     run                     Starts Che application server in current console
     start                   Starts Che application server in new console
     stop                    Stops Che application server

The -r flag sets the DOCKER_MACHINE_HOST system environment variable. Set this to the IP address of the node
that is running your Docker daemon. Only necessary to set this if on Linux and your browser clients are not
localhost, ie they are remote. This property automatically set for Che on Windows and Mac.

!!! Could not find Docker client. Expected at Windows: %DOCKER_TOOLBOX_INSTALL_PATH%\docker.exe, Mac: /usr/local/bin/docker, Linux: /usr/bin/docker.

Looks like something went wrong. Possible issues:
1. (Win | Mac) VirtualBox not installed          ==> Rerun Docker Toolbox installation
2. (Win | Mac) Docker Machine not installed      ==> Rerun Docker Toolbox installation
3. (Win | Mac) Docker is not reachable           ==> Docker VM failed to start
4. (Win | Mac) Docker ok, but docker ps fails    ==> Docker environment variables not set properly
5. (Linux) Docker is not reachable               ==> Install: wget -qO- https://get.docker.com/ | sh
6. (Linux) Permissions not properly set          ==> Che must run as UID 1000 with user in docker group
7. Could not find the Che app server             ==> Did /tomcat get moved away from CHE_HOME?
8. Wrong version of Java found                   ==> Che requires Java 1.8
9. Did you use the right parameter syntax?       ==> See usage

We have seen issues with VirtualBox on Windows where your VM gets corrupted when your computer is
suspended while the VM is still running. This will appear as SSH or ethernet connection issues. This is
rare, but if encountered, current known solution is to uninstall VirtualBox and Docker Toolbox, and then
reinstall.

Che Environment Variables:
  (REQUIRED) JAVA_HOME                             ==> Location of Java runtime
  (REQUIRED: WIN|MAC) DOCKER_TOOLBOX_INSTALL_PATH  ==> Location of Docker Toolbox
  (REQUIRED: WIN|MAC) VBOX_MSI_INSTALL_PATH        ==> Location of VirtualBox
  (OPTIONAL) CHE_HOME                              ==> Directory where Che is installed
  (OPTIONAL) CHE_LOCAL_CONF_DIR                    ==> Directory with custom Che .properties files
  (OPTIONAL) CHE_LOGS_DIR                          ==> Directory for Che output logs
  (OPTIONAL) CHE_DOCKER_MACHINE_NAME               ==> (Win | Mac) Name of VM created by docker-machine
  (OPTIONAL) DOCKER_MACHINE_HOST                   ==> (Linux) Docker host IP - set if browser clients remote

Usage:
  che [OPTIONS] [run | start | stop]
     -i,      --image        Launches Che within a Docker container using latest image
     -i:tag,  --image:tag    Launches Che within a Docker container using specific image tag
     -p:port, --port:port    Port that Che server will use for HTTP requests; default=8080
     -r:ip,   --remote:ip    If Che clients are not localhost, set to IP address of Che server
     -m:vm,   --machine:vm   For Win & Mac, sets the docker-machine VM name to vm; default=default
     -s,      --suppress     Do not print browser client connection information
     -h,      --help         Show this help
     -d,      --debug        Use debug mode (prints command line options + app server debug)
     run                     Starts Che application server in current console
     start                   Starts Che application server in new console
     stop                    Stops Che application server

The -r flag sets the DOCKER_MACHINE_HOST system environment variable. Set this to the IP address of the node
that is running your Docker daemon. Only necessary to set this if on Linux and your browser clients are not
localhost, ie they are remote. This property automatically set for Che on Windows and Mac.

**Comment by Cookie-Wang** [comment]
should install docker? in linux

**Comment by Cookie-Wang** [comment]
who has detail to install eclipse che?contain how to  config docker!  thanks

**Comment by TylerJewell** [comment]
​The user which you started Che is not part of the linux user group
"docker".  We require this for permissions reasons.

You must install and configure docker and then have your user in the right
group.

!!! This Linux user is not in docker group See
https://docsdockercom/engine/installation/ubuntulinux/#create-a-docker-group
​

Tyler Jewell | CEO | tyler@​codenvy.​com | 9​78​.8​84​.53​55

**Comment by webmutation** [comment]
Hi, I am running as root on Ubuntu 14.04.3 LTS (on softlayer cloud)
uid=0(root) gid=0(root) groups=0(root)

and it gives me the following error

"!!! This Linux user was launched with a UID != 1000. Che must run under UID 1000. See https://eclipse-che.readme.io/docs/usage#section-cannot-create-projects
!!!"

the docs need to be more explicit in this regard, since as root I have all the required permissions. shouldn't you be checking for permissions instead of a specific UID?

**Comment by TylerJewell** [comment]
@webmutation Oops - we changed the docs this weekend, and the URL from the che script is pointing to the wrong place.  The link should now go [here](https://eclipse-che.readme.io/docs/usage#section-cannot-create-projects).  We explain the need for UID and why it happens here.

You can skip the UID check on the command line of che with the --skip options.

**Comment by TylerJewell** [comment]
@webmutation This link should be fixed in the docs.  This way the product doesn't point to an improper location.

**Comment by yuankaiz** [comment]
@TylerJewell the link you give does not work, please fix. 
Also, I've add my user in docker group, but it insists I need UID 1000 to run. 
I could get Che to run by -s:uid, but that way I cannot create any project in the workspace. 
Why Che want us to run with UID 1000 here?

**Comment by TylerJewell** [comment]
@yuankaiz - the OP link now works and I updated my link as well.  The explanation for why UID 1000 is needed is on this doc page in the link.  @eivantsov can explain more details as well.

**Comment by rajasekaran07** [comment]
@TylerJewell 

I have added my user into docker group but still I am getting the error.

root@ipaclient:~# adduser raja
Adding user `raja' ...
Adding new group`raja' (1001) ...
Adding new user `raja' (1001) with group`raja' ...
Creating home directory `/home/raja' ...
Copying files from`/etc/skel' ...
New password:
Retype new password:
passwd: password updated successfully
Changing the user information for raja
Enter the new value, or press ENTER for the default
        Full Name []: Rajasekaran
        Room Number []:
        Work Phone []:
        Home Phone []:
        Other []:
Is the information correct? [Y/n] y
root@ipaclient:~# usermode -aG docker raja
No command 'usermode' found, did you mean:
 Command 'usermod' from package 'passwd' (main)
usermode: command not found
root@ipaclient:~# usermod -aG docker raja
root@ipaclient:~# su raja
raja@ipaclient:/root$ cd /home/user/eclipse-che-4.3.0-RC1-SNAPSHOT/bin/
raja@ipaclient:/home/user/eclipse-che-4.3.0-RC1-SNAPSHOT/bin$ ./che.sh run

!!!
!!! This Linux user was launched with a UID != 1000. Che must run under UID 1000. See https://eclipse-che.readme.io/docs/usage#section-cannot-create-projects
!!!

Kindly help on this...
Regards,
Raja

**Comment by ddementieva** [comment]
@rajasekaran07 Your userID is 1001, not 1000. You can skip ID with `--skip:uid` parameter when starting Che.

**Comment by rajasekaran07** [comment]
@TylerJewell @ddementieva 
--skip:uid also not working...

root@ipaclient:~# /home/user/eclipse-che-4.3.0-RC1-SNAPSHOT/bin/./che.sh --skip:uid

############## HOW TO CONNECT YOUR CHE CLIENT ###############
After Che server has booted, you can connect your clients by:
1. Open browser to http://localhost:8080, or:
2. Open native chromium app.
#############################################################

/home/user/eclipse-che-4.3.0-RC1-SNAPSHOT/bin/./che.sh: line 576: /home/user/eclipse-che-4.3.0-RC1-SNAPSHOT/tomcat/bin/catalina.sh: Permission denied

Should we install tomcat before starting che?

Regards,
Raja

**Comment by ddementieva** [comment]
@rajasekaran07 Lets track this issue https://github.com/eclipse/che/issues/2368

**Comment by rajasekaran07** [comment]
Sure... Sorry for duplication..

**Comment by sqs** [comment]
I experienced this issue and was able to fix it.

When I ran `che start`, I was consistently seeing this error:

```
$ che start
WARNING: No swap limit support
INFO: ECLIPSE CHE: Found image codenvy/che-server:latest
INFO: ECLIPSE CHE: Starting container...
INFO: ECLIPSE CHE: Server logs at "docker logs -f che-server"
INFO: ECLIPSE CHE: Server booting...
---------------------------------------
ERROR: !!!
ERROR:      ECLIPSE CHE: Timeout waiting for server. Run "docker logs che-server" to inspect the issue.
ERROR: !!!
---------------------------------------
```

The `docker logs che-server` output was:

```
$ docker logs -f che-server

!!!
!!! This Linux user is not in 'docker' group. See https://docs.docker.com/engine/installation/ubuntulinux/#create-a-docker-group
!!!
```

To fix it, I ran `che start` with `CHE_USER=1000` (my local user's uid). I got the same `che start` output as above, but this changed the `docker logs che-server` output to:

```
$ docker logs -f che-server

!!!
!!! Running 'docker' succeeded, but 'docker ps' failed. 
The file /var/run/docker.sock does not have appropriate permissions. 
OWNER READ:  r 
OWNER WRITE: w 
GROUP READ:  r 
GROUP WRITE: w 
OTHER READ:  - 
OTHER WRITE: - 
Run 'sudo chmod 666 /var/run/docker.sock' to give the right permissions.
!!!
```

After running `sudo chmod 666 /var/run/docker.sock` as it told me to, `che start` works:

```
$ CHE_USER=1000 che start
WARNING: No swap limit support
INFO: ECLIPSE CHE: Found image codenvy/che-server:latest
INFO: ECLIPSE CHE: Starting container...
INFO: ECLIPSE CHE: Server logs at "docker logs -f che-server"
INFO: ECLIPSE CHE: Server booting...
INFO: ECLIPSE CHE: Booted and reachable
INFO: ECLIPSE CHE: Ver: 4.7.2
INFO: ECLIPSE CHE: Use: http://172.17.0.1:9090
INFO: ECLIPSE CHE: API: http://172.17.0.1:9090/swagger
```

**Comment by anshada** [comment]
@sqs thanks, this fixed the issue
